torch/csrc/distributed/c10d/Backoff.cpp

```diff
@@ -0,0 +1,75 @@
+#include <torch/csrc/distributed/c10d/Backoff.hpp>
+
+#include <exception>
+#include <stdexcept>
+
+namespace c10d {
+namespace {
+constexpr std::chrono::milliseconds kZeroInterval{0};
+
+int32_t randSeed() {
+  std::random_device rd;
+  return rd();
+}
+} // namespace
+
+ExponentialBackoff::ExponentialBackoff() : gen_(randSeed()) {}
```

Comment:
Should we name it ExponentialBackoff[AndJ|With]itter?

Reply:
This is loosely following the naming schemes from the very popular Golang backoff library https://pkg.go.dev/github.com/cenkalti/backoff/v4#ExponentialBackOff

I do think any good exp backoff implementation will have jitter but might be more clear for people who aren't familiar with this style of distributed programming

Reply:
renamed